internal/controllers/provisioningrequest_controller.go

```diff
@@ -1172,6 +1174,15 @@ func (r *ProvisioningRequestReconciler) handleProvisioningRequestDeletion(
 		return false, fmt.Errorf("failed to list namespaces: %w", err)
 	}
 	for _, ns := range namespaceList.Items {
+		// Workaround to deal with stuck ObservabilityAddon finalizer
+		// Check for ObservabilityAddon CR and remove finalizers before deleting namespace
+		if err := r.handleObservabilityAddonCleanup(ctx, ns.Name); err != nil {
+			r.Logger.WarnContext(ctx, "Failed to handle ObservabilityAddon cleanup",
+				slog.String("namespace", ns.Name),
+				slog.String("error", err.Error()))
+			// Continue with deletion attempt even if cleanup fails
+		}
```

Comment:
i think here a little more safeguard would make sense instead of brute forcing this at every requeue...something like....
```
 // Only remove finalizers if deletion timestamp is set and older than threshold
  if ns.DeletionTimestamp.IsZero() &&
     time.Since(ns.DeletionTimestamp.Time) > time.Minute*5 {
      // Safe to remove stuck finalizers
  }
```